pydicom/values.py

```diff
@@ -332,62 +379,78 @@ def convert_numbers(byte_string, is_little_endian, struct_format):
     if length % bytes_per_value != 0:
         logger.warning("Expected length to be even multiple of number size")
 
-    format_string = "%c%u%c" % (endianChar, length // bytes_per_value,
-                                struct_format)
-
-    value = unpack(format_string, byte_string)
+    format_string = f"{endianChar}{length // bytes_per_value}{struct_format}"
+    value: Union[Tuple[int, ...], Tuple[float, ...]] = (
+        unpack(format_string, byte_string)
+    )
 
     # if the number is empty, then return the empty
     # string rather than empty list
     if len(value) == 0:
         return ''
-    elif len(value) == 1:
+
+    if len(value) == 1:
         return value[0]
-    else:
-        # convert from tuple to a list so can modify if need to
-        return list(value)
+
+    # convert from tuple to a list so can modify if need to
+    return list(value)
 
 
-def convert_OBvalue(byte_string, is_little_endian, struct_format=None):
-    """Return encoded 'OB' value as :class:`bytes` or :class:`str`."""
+def convert_OBvalue(
+    byte_string: bytes,
+    is_little_endian: bool,
+    struct_format: Optional[str] = None
+) -> bytes:
+    """Return encoded 'OB' value as :class:`bytes`."""
     return byte_string
 
```

Comment:
Do we still need this function?

Reply:
Technically no, all the O* VRs could just be added as a pass through to `convert()`, but there's no harm in having them monkey patchable in case users want to implement byte swapping for big endian or something.  

Although we could probably just use `None` in `converters` and check to see if the user has implemented it. Maybe in another PR?

Reply:
Should we implement the byte-swapping for v3?  It seems a long-standing "laziness" that perhaps should be dealt with properly.  Perhaps needs discussion in another issue/PR (after v2.1) and if so, then should leave these in for now.